### ISSUE TITLE
Error when linking executable

### ISSUE THREAD

**Issue by Phate6660** [nikp123/xava#23]
```
Scanning dependencies of target iniparser
[ 15%] Building C object CMakeFiles/iniparser.dir/lib/iniparser/src/dictionary.c.o
[ 15%] Building C object CMakeFiles/iniparser.dir/lib/iniparser/src/iniparser.c.o
[ 23%] Linking C static library libiniparser.a
[ 23%] Built target iniparser
Scanning dependencies of target xava
[ 46%] Building C object CMakeFiles/xava.dir/src/input/alsa.c.o
[ 46%] Building C object CMakeFiles/xava.dir/src/output/graphical_x.c.o
[ 46%] Building C object CMakeFiles/xava.dir/src/output/graphical_sdl.c.o
[ 53%] Building C object CMakeFiles/xava.dir/src/input/fifo.c.o
[ 61%] Building C object CMakeFiles/xava.dir/src/misc/inode_watcher.c.o
[ 69%] Building C object CMakeFiles/xava.dir/src/output/graphical.c.o
[ 76%] Building C object CMakeFiles/xava.dir/src/shared.c.o
[ 84%] Building C object CMakeFiles/xava.dir/src/config.c.o
[ 92%] Building C object CMakeFiles/xava.dir/src/xava.c.o
[100%] Linking C executable xava
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: CMakeFiles/xava.dir/src/output/graphical_x.c.o: in function `apply_window_settings_x':
graphical_x.c:(.text+0x10bd): undefined reference to `glXSwapIntervalEXT'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/xava.dir/build.make:205: xava] Error 1
make[1]: *** [CMakeFiles/Makefile2:78: CMakeFiles/xava.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

**Comment by nikp123** [comment]
Since you're on Gentoo, may you check if your libGL.so (or similar) features the symbol "glXSwapIntervalEXT". Try to enabled it in your build config if it's not there.

I'm sorry, but I can't provide exact support for Gentoo, since the builds vary wildly (I'm sure you already know that)

**Comment by nikp123** [comment]
And seeing the issues that are gentoo specific, this seems like a OS issue more than anything. There isn't any valid alternative to the syscall "glXSwapIntervalEXT".

**Comment by nikp123** [comment]
Did you try rebuilding mesa?
By looking into Gentoo forums, it seems like the only culprit.
Cannot test since I don't feel like setting up Gentoo for just an bug report.

**Comment by Phate6660** [comment]
I have checked my `libGL.so` and apparently it doesn't contain `glXSwapIntervalEXT` for some reason.

I have rebuilt mesa actually a few times already because I installed a new GPU and needed to rebuild it for the drivers, but it still doesn't have it.

I can only guess that I'm missing a USE flag somewhere for mesa to have that api, I just don't know what.

I'll close this since it seems to be an issue with either me, or Gentoo.
